nemo/collections/asr/models/classification_models.py

```diff
@@ -489,95 +647,109 @@ def _update_decoder_config(self, cfg):
 
         OmegaConf.set_struct(cfg, True)
 
-    def _setup_transcribe_dataloader(self, config: Dict) -> 'torch.utils.data.DataLoader':
-        """
-        Setup function for a temporary data loader which wraps the provided audio file.
 
-        Args:
-            config: A python dictionary which contains the following keys:
-            paths2audio_files: (a list) of paths to audio files. The files should be relatively short fragments. \
-                Recommended length per file is between 5 and 25 seconds.
-            batch_size: (int) batch size to use during inference. \
-                Bigger will result in better throughput performance but would use more memory.
-            temp_dir: (str) A temporary directory where the audio manifest is temporarily
-                stored.
+class EncDecRegressionModel(_EncDecBaseModel):
+    """Encoder decoder class for speech regression models.
+    Model class creates training, validation methods for setting up data
+    performing model forward pass.
+    """
 
+    @classmethod
+    def list_available_models(cls) -> List[PretrainedModelInfo]:
+        """
+        This method returns a list of pre-trained model which can be instantiated directly from NVIDIA's NGC cloud.
         Returns:
-            A pytorch DataLoader for the given audio file(s).
+            List of available pre-trained models.
         """
-        dl_config = {
-            'manifest_filepath': os.path.join(config['temp_dir'], 'manifest.json'),
-            'sample_rate': self.preprocessor._sample_rate,
-            'labels': self.cfg.labels,
-            'batch_size': min(config['batch_size'], len(config['paths2audio_files'])),
-            'trim_silence': False,
-            'shuffle': False,
+        result = []
+
+        return result
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        OmegaConf.set_struct(cfg, False)
+        cfg.is_regression_task = True
+        OmegaConf.set_struct(cfg, True)
+        super().__init__(cfg=cfg, trainer=trainer)
+
+    def _setup_preprocessor(self):
+        return EncDecRegressionModel.from_config_dict(self._cfg.preprocessor)
+
+    def _setup_encoder(self):
+        return EncDecRegressionModel.from_config_dict(self._cfg.encoder)
+
+    def _setup_decoder(self):
+        return EncDecRegressionModel.from_config_dict(self._cfg.decoder)
+
+    def _setup_loss(self):
+        return MSELoss()
+
+    def _setup_metrics(self):
+        pass
```

Comment:
I believe PTL supports MSE as a metric yes? Can we add that here?

Reply:
Sure, indeed I'm adding also MAE as it's a popular metric for regression. MSE will be the same as loss for this model, but it's possible to create regression models with other loss functions.

Just one comment on this. The loss that is computed and the end of validation/testing is the average loss of the losses computed for every batch, which is not the same as the total loss for the validation/test dataset. In the case of metrics, I'm doing the later and the output is the actual MSE/MAE for the complete dataset.